core/src/main/java/org/mule/runtime/core/internal/metadata/MuleMetadataManager.java

```diff
@@ -170,7 +170,7 @@ public MetadataCache getMetadataCache(String id) {
     try {
       return producer.get();
     } catch (InvalidComponentIdException e) {
-      return failure(e);
+      return failure(null, e.getMessage(), FailureCode.COMPONENT_NOT_FOUND, e.getDetailedMessage());
```

Comment:
add a new failure builder that accepts only an exception and an overrided failure code